pkg/syncer/spec/spec_controller.go

```diff
@@ -157,6 +151,22 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, workloadClusterLogicalClu
 		klog.V(2).InfoS("Set up downstream informer", "clusterName", workloadClusterLogicalClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
 	}
 
+	secretMutator := specmutators.NewSecretMutator()
+
+	secretIndexer := upstreamInformers.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Informer().GetIndexer()
+	deploymentMutator := specmutators.NewDeploymentMutator(upstreamURL, secretIndexer, workloadClusterLogicalClusterName, byWorkspace)
```

Comment:
am against this. Why do we break the patterns we have prepared in the code and build one-off code again?